flow-api-tutorial/README.md

```diff
@@ -0,0 +1,334 @@
+[![Jina Docs](https://github.com/jina-ai/jina/blob/master/.github/badges/docs-badge.svg "Checkout our docs and learn Jina")](https://docs.jina.ai)
+[![We are hiring](https://github.com/jina-ai/jina/blob/master/.github/badges/jina-corp-badge-hiring.svg "We are hiring full-time position at Jina")](https://jobs.jina.ai)
+<a href="https://twitter.com/intent/tweet?text=%F0%9F%91%8DCheck+out+Jina%3A+the+New+Open-Source+Solution+for+Neural+Information+Retrieval+%F0%9F%94%8D%40JinaAI_&url=https%3A%2F%2Fgithub.com%2Fjina-ai%2Fjina&hashtags=JinaSearch&original_referer=http%3A%2F%2Fgithub.com%2F&tw_p=tweetbutton" target="_blank">
+  <img src="https://github.com/jina-ai/jina/blob/master/.github/badges/twitter-badge.svg"
+       alt="tweet button" title="👍Share Jina with your friends on Twitter"></img>
+</a>
+[![Python 3.7 3.8](https://github.com/jina-ai/jina/blob/master/.github/badges/python-badge.svg "Jina supports Python 3.7 and above")](#)
+[![Docker](https://github.com/jina-ai/jina/blob/master/.github/badges/docker-badge.svg "Jina is multi-arch ready, can run on differnt architectures")](https://hub.docker.com/r/jinaai/jina/tags)
+
+</p>
+
+In this demo, we'll use some code snippets to show you how to use flow API for indexing/searching different data. Make sure you've gone through [Jina 101](https://github.com/jina-ai/jina/tree/master/docs/chapters/101) and understood the Pod, Flow, Yaml before moving on. 
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [3 APIs for indexing and searching your data](#3-APIs-for-indexing-and-searching-your-data)
+    - [ndarray API](#1-ndarray-api)
+    - [files API](#2-files-api)
+    - [lines API](#3-lines-api)
+- [Wrap up](#wrap-up)
+- [Next Steps](#next-steps)
+- [Documentation](#documentation)
+- [Community](#community)
+- [License](#license)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+## Overview
+Jina supports you to index/search your data in a simple way. And it exposes 3 APIs for each of them, which help you to index or search `np.ndarray`, `files`, and `text`.
+
+## 3 APIs for indexing and searching your data
+
+### 1. ndarray API
+`index_ndarray()` is the API for indexing `np.ndarray` 
+
+```python
+import numpy as np
+from jina.flow import Flow
+input_data = np.random.random((3,8))
+f = Flow().add(uses='_logforward')
+with f:
+    f.index_ndarray(input_data)
+```
+    
+* Add an empty pod with config `_logforward` to the Flow. [`_logforward`](https://docs.jina.ai/chapters/simple_exec.html) is a built-in yaml, which just forwards input data to the results, and it locates in `jina/resources/executors._forward.yml`. You can also use your own [yaml](https://docs.jina.ai/chapters/yaml/yaml.html) to organize `pods`.
```

Comment:
empty -> dummy

Reply:
which just forwards input data to the results,

 which just forwards input message and print it to the log